api/column_family.hh

```diff
@@ -68,11 +68,10 @@ struct map_reduce_column_families_locally {
     std::function<std::unique_ptr<std::any>(std::unique_ptr<std::any>, std::unique_ptr<std::any>)> reducer;
     future<std::unique_ptr<std::any>> operator()(replica::database& db) const {
         auto res = seastar::make_lw_shared<std::unique_ptr<std::any>>(std::make_unique<std::any>(init));
-        for (auto& i: db.get_column_families()) {
+        auto column_families = db.get_column_families();
```

Comment:
The yield is there to prevent stalls. Now the copy can stall.

Reply:
While looking for a way to fix, I saw this:

```c++
void database::add_column_family(keyspace& ks, schema_ptr schema, column_family::config cfg) {
    schema = local_schema_registry().learn(schema);
    schema->registry_entry()->mark_synced();
    // avoid self-reporting
    auto& sst_manager = is_system_table(*schema) ? get_system_sstables_manager() : get_user_sstables_manager();
    lw_shared_ptr<column_family> cf;
    if (cfg.enable_commitlog && _commitlog) {
        db::commitlog& cl = schema->ks_name() == db::schema_tables::NAME && _uses_schema_commitlog
                ? *_schema_commitlog
                : *_commitlog;
        cf = make_lw_shared<column_family>(schema, std::move(cfg), cl, _compaction_manager, sst_manager, *_cl_stats, _row_cache_tracker);
    } else {
       cf = make_lw_shared<column_family>(schema, std::move(cfg), column_family::no_commitlog(), _compaction_manager, sst_manager, *_cl_stats, _row_cache_tracker);
    }
    cf->set_durable_writes(ks.metadata()->durable_writes());

    auto uuid = schema->id();
    if (_column_families.contains(uuid)) {
        throw std::invalid_argument("UUID " + uuid.to_sstring() + " already mapped");
    }
    auto kscf = std::make_pair(schema->ks_name(), schema->cf_name());
    if (_ks_cf_to_uuid.contains(kscf)) {
        throw std::invalid_argument("Column family " + schema->cf_name() + " exists");
    }
    ks.add_or_update_column_family(schema);
    cf->start();
    _column_families.emplace(uuid, std::move(cf));
    _ks_cf_to_uuid.emplace(std::move(kscf), uuid);
    if (schema->is_view()) {
        find_column_family(schema->view_info()->base_id()).add_or_update_view(view_ptr(schema));
    }
}
```

If an exception happens during _ks_cf_to_uuid.emplace() (or a previous call), we won't undo ks.add_or_update_column_family().

Reply:
Anyway, to fix, we should add a lock around table addition (not easy, since it requires futurizing that code path).